src/robot/running/keywordrunner.py

```diff
@@ -113,7 +114,12 @@ def _get_and_report_failure(self):
         return failure
 
 
-class ForLoopRunner(object):
+class BasicForLoopRunner(object):
+    """A basic For Loop.
+
+    To make a new kind of For loop, subclass this and override
+    _flavor_name() and _transform_items(items).
+    """
```

Comment:
It feels a bit dirty, but the best response here may be to delete the docstring.